galsim/phase_screens.py

```diff
@@ -53,6 +53,79 @@ def __calcOptStepK(lam, diam, obscuration, gsparams):
 _GSScreenShare = {}
 
 
+def writeScreenShare(filename):
+    """Write `AtmosphericScreen` shared memory to a file.  See discussion in `AtmosphericScreen`
+    docstring for more information.
+
+    Parameters:
+        filename: Filename to which to write.
+    """
+    import pickle
+    screenShare = {}
+    for k, v in _GSScreenShare.items():
+        oneScreen = {}
+        oneScreen['f'] = np.frombuffer(v['f'], dtype=np.float64)
+        oneScreen['x'] = np.frombuffer(v['x'], dtype=np.float64)
+        oneScreen['y'] = np.frombuffer(v['y'], dtype=np.float64)
+        oneScreen['alpha'] = v['alpha'].value
+        oneScreen['kmin'] = v['kmin'].value
+        oneScreen['kmax'] = v['kmax'].value
+        oneScreen['x0'] = v['x0'].value
+        oneScreen['y0'] = v['y0'].value
+        oneScreen['xperiod'] = v['xperiod'].value
+        oneScreen['yperiod'] = v['yperiod'].value
+        oneScreen['instantiated'] = v['instantiated'].value
+        oneScreen['ctx_name'] = v['ctx_name'].value
+        screenShare[k] = oneScreen
```

Comment:
@jchiang87's request would be easy to implement by making the above part of this function be in a method called perhaps `getScreenShare`.  Then this function would be just
```
with open(filename, 'wb') as f:
        pickle.dump(self.getScreenShare(), f)
```
Likewise for the read function, most of it could be in `setScreenShare` or some similar name.

Reply:
Jim's proposal [here](https://github.com/LSSTDESC/imSim/pull/235#discussion_r341879513) looks good.  One modification I'd like to propose: instead of using screen `get_picklable()` and `set_data()`, I think I can work things out so that the screens get included in the psf pickle automagically if the pickle is created with in a specific context manager (which temporarily sets a global GalSim pickling variable).  Then when reading back in, the `AtmosphericScreen.__setstate__` method would also load the screen if it happens to find it in the pickle.  E.g.,

```python
pickle.dump(psf1, file1)  # Only saves psf1, no shared screens
with galsim.pickleShared():
    pickle.dump(psf2, file2) # Saves both psf2 and shared screens, but only shared screens associated with psf2

psf1b = pickle.load(file1)  # Works if psf1 shared screens are already loaded
psf2b = pickle.load(file2)  # Works whether or not psf2 shared screens are already loaded
```

Reply:
That's cute.  :)  Seems like a nice API.